// discord.bio/dist/structures/Activity.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const discord_js_1 = require("discord.js");
const Emoji_1 = __importDefault(require("./Emoji"));
const RichPresenceAssets_1 = __importDefault(require("./RichPresenceAssets"));
class Activity {
    constructor(bio, data) {
        const { name, assets, createdTimestamp, timestamps, state, type, url, party, emoji, details, flags, applicationID } = data;
        this.emoji = emoji ? new Emoji_1.default(bio, emoji) : null;
        this.name = name;
        this.assets = assets ? new RichPresenceAssets_1.default(this, assets) : null;
        this.createdTimestamp = createdTimestamp;
        timestamps ? this.timestamps = {
            start: timestamps.start ? new Date(timestamps.start) : null,
            end: timestamps.end ? new Date(timestamps.end) : null,
        } : this.timestamps = null;
        this.details = details;
        this.state = state;
        this.type = type;
        this.url = url;
        this.party = party;
        this.applicationID = applicationID;
        this.flags = new discord_js_1.ActivityFlags(flags);
    }
    /**
     * Whether this activity is equal to another activity.
     * @param activity The actvity to compare with
     */
    equals(activity) {
        return (this === activity ||
            (activity && this.name === activity.name && this.type === activity.type && this.url === activity.url));
    }
    /**The time the activity was created at */
    get createdAt() {
        return new Date(this.createdTimestamp);
    }
}
module.exports = Activity;
