// discord.bio/dist/structures/RichPresenceAssets.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const enumerable_1 = __importDefault(require("../util/enumerable"));
class RichPresenceAssets {
    constructor(activity, assets) {
        this.activity = activity;
        this.largeText = assets.largeText || null;
        this.smallText = assets.largeText || null;
        this.largeImage = assets.largeImage || null;
        this.smallImage = assets.smallImage || null;
    }
    /**Gets the URL of the large image asset */
    largeImageURL({ format, size } = {}) {
        if (!this.largeImage)
            return null;
        if (/^spotify:/.test(this.largeImage)) {
            return `https://i.scdn.co/image/${this.largeImage.slice(8)}`;
        }
        else if (/^twitch:/.test(this.largeImage)) {
            return `https://static-cdn.jtvnw.net/previews-ttv/live_user_${this.largeImage.slice(7)}.png`;
        }
        return `https://cdn.discordapp.com/app-assets/${this.activity.applicationID}/${this.largeImage}${format ? `.${format}` : ''}${size ? `?size=${size}` : ''}`;
    }
    /**Gets the URL of the small image asset */
    smallImageURL({ format, size } = {}) {
        if (!this.smallImage)
            return null;
        else
            return `https://cdn.discordapp.com/app-assets/${this.activity.applicationID}/${this.smallImage}${format ? `.${format}` : ''}?size=${size}`;
    }
}
__decorate([
    enumerable_1.default(false)
], RichPresenceAssets.prototype, "activity", void 0);
module.exports = RichPresenceAssets;
