// discord.bio/dist/structures/User.js
"use strict";
const discord_js_1 = require("discord.js");
/**Represent A User */
class User {
    /**
     * @param rawUser Thee raw user returned by the API
     */
    constructor(rawUser) {
        const { id, username, avatar, discriminator, public_flags } = rawUser;
        this.id = id;
        this.username = username;
        this.discriminator = discriminator;
        this.avatar = avatar;
        this.tag = `${this.username}#${this.discriminator}`;
        this.public_flags = new discord_js_1.UserFlags(public_flags);
        const a = "https://discord.com/assets/";
        const urls = [
            "6debd47ed13483642cf09e832ed0bc1b.png",
            "322c936a8c8be1b803cd94861bdfa868.png",
            "dd4dbc0016779df1378e7812eabaa04d.png",
            "0e291f67c9274a1abdddeb3fd919cbaa.png",
            "1cbd08c76f8af6dddce02c5138971129.png"
        ].map(x => a + x);
        this.defaultAvatarURL = urls[parseInt(this.discriminator.slice(3))] || urls[parseInt(this.discriminator.slice(3)) - 5];
    }
    /**
     * The link to the user's avatar
     * @param options Options for the image
     */
    avatarURL(options = { format: 'webp', dynamic: false }) {
        options.format ? {} : options.format = 'webp';
        let url = `https://cdn.discordapp.com/avatars/${this.id}/${this.avatar}`;
        if (!this.avatar)
            return null;
        if (!this.avatar.startsWith('a_') && options.format === 'gif')
            return null;
        else
            url += '.' + options.format;
        if (options.size)
            url += `?size=${options.size.toString()}`;
        return url;
    }
    /**
     * The link to the user's avatar if they have one,or to there defualt one if they don't have an avatar.
     * @warn Notice: Default avatar has a fixed size and is always a .png image
     * @param options Options for the image
     */
    displayAvatarURL(options) {
        if (this.avatarURL(options))
            return this.avatarURL(options);
        else
            return this.defaultAvatarURL;
    }
}
module.exports = User;
