// discord.bio/dist/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const User_1 = __importDefault(require("./structures/User"));
exports.User = User_1.default;
const RawUser_1 = __importDefault(require("./structures/RawUser"));
exports.RawUser = RawUser_1.default;
const details_1 = __importDefault(require("./endpoints/user/details"));
const search_1 = __importDefault(require("./endpoints/user/search"));
const APIVersion_1 = __importDefault(require("./endpoints/APIVersion"));
const topUpvoted_1 = __importDefault(require("./endpoints/topUpvoted"));
const presence_1 = __importDefault(require("./endpoints/user/presence"));
const api_1 = __importDefault(require("./util/api"));
const discord_js_1 = require("discord.js");
exports.UserFlags = discord_js_1.UserFlags;
exports.Collection = discord_js_1.Collection;
const events_1 = require("events");
const enumerable_1 = __importDefault(require("./util/enumerable"));
const DBioAPIError_1 = __importDefault(require("./structures/DBioAPIError"));
exports.DBioAPIError = DBioAPIError_1.default;
const Constants_1 = require("./util/Constants");
const util_1 = require("util");
/**The main hub for interacting with the discord.bio API. */
class Bio extends events_1.EventEmitter {
    /**
     * @param baseURL - The API base URL
     */
    constructor(baseURL) {
        super();
        this.baseURL = baseURL || Constants_1.defaults.baseurl;
        this.__limit = 100;
        this.APIVersion = APIVersion_1.default;
        this.topUpvoted = topUpvoted_1.default;
        //semver
        this.totalUsers = util_1.deprecate(function () { return 1; }, 'This endpoint no longer exists');
        this.api = api_1.default;
        this.users = {
            bio: this,
            details: details_1.default,
            search: search_1.default,
            presence: presence_1.default
        };
        this.bio = this;
        this.__quota = this.__limit;
        this.__quota_reset = Date.now();
        this.__outgoing_requests = 0;
        this.APIVersion();
    }
}
__decorate([
    enumerable_1.default(false)
], Bio.prototype, "api", void 0);
__decorate([
    enumerable_1.default(false)
], Bio.prototype, "bio", void 0);
exports.Bio = Bio;
